### ISSUE TITLE
 bundling failed: Error: Unable to resolve module `../Components/AccessibilityInfo/AccessibilityInfo` from `node_modules/react-native/Libraries/react-native/react-native-implementation.js`: 

### ISSUE THREAD

**Issue by sangeethkumarv** [microsoft/react-native-macos#564]
While running the app in bundler it fails with the following message in the terminal

warning: the transform cache was reset.
Loading dependency graph, done.

To reload the app press "r"
To open developer menu press "d"

error: bundling failed: Error: Unable to resolve module `../Components/AccessibilityInfo/AccessibilityInfo` from `node_modules/react-native/Libraries/react-native/react-native-implementation.js`: 

None of these files exist:
  * node_modules/react-native/Libraries/Components/AccessibilityInfo/AccessibilityInfo(.native|.macos.js|.native.js|.js|.macos.json|.native.json|.json|.macos.ts|.native.ts|.ts|.macos.tsx|.native.tsx|.tsx)
  * node_modules/react-native/Libraries/Components/AccessibilityInfo/AccessibilityInfo/index(.native|.macos.js|.native.js|.js|.macos.json|.native.json|.json|.macos.ts|.native.ts|.ts|.macos.tsx|.native.tsx|.tsx)
    at ModuleResolver.resolveDependency (/Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:163:15)
    at ResolutionRequest.resolveDependency (/Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/node-haste/DependencyGraph/ResolutionRequest.js:52:18)
    at DependencyGraph.resolveDependency (/Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/node-haste/DependencyGraph.js:282:16)
    at Object.resolve (/Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/lib/transformHelpers.js:267:42)
    at dependencies.map.result (/Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/DeltaBundler/traverseDependencies.js:426:31)
    at Array.map (<anonymous>)
    at resolveDependencies (/Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/DeltaBundler/traverseDependencies.js:423:18)
    at /Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/DeltaBundler/traverseDependencies.js:275:33
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/dev/Desktop/Sangeeth/mytakeaway2.0/Myt-Mac/MYT_MacOS/node_modules/metro/src/DeltaBundler/traverseDependencies.js:87:24)
 BUNDLE  [macos, dev] ./index.js ░░░░░░░░░░░░░░░░ 3.1% (9/77), failed.

**Comment by sangeethkumarv** [comment]
I fixed this by removing .ios.js extension with .js in files inside react-native/libraries/components

For example, I changed AccessibilityInfo.ios.js --> AccessibilityInfo.js
Likewise, there are so many files which need these changes like Checkbox, DatePicker, Alert, Settings, etc

**Comment by acoates-ms** [comment]
The above error indicates that your metro config is not correct.  It should never be loaded files from react-native when bundling for macos -- it should get all the files from react-native-macos.

Are you on 0.62? And using at least version 4.9 of @react-native-community/cli?

**Comment by acoates-ms** [comment]
For 0.61 you'd need a seperate metro config for macos, like react-native-netinfo does:
https://github.com/react-native-community/react-native-netinfo/blob/master/metro.config.macos.js

**Comment by Abhishek12345679** [comment]
what is causing this issue and how do i fix this ?

**Comment by Cedrigax** [comment]
Close running Metro and run `react-native run-macos` again. It'll solve the problem. No need to change or rename any file

**Comment by acoates-ms** [comment]
When bundling for macos, metro should be told to redirect any react-native import to react-native-macos.  

In 0.61 you have to do this manually by changing your config when bundling for macos.  (The config wont work for other platforms)
```js
  resolver: {
    extraNodeModules: {
      'react-native': path.resolve(__dirname, 'node_modules/react-native-macos')
    },
  },
  serializer: {
    getModulesRunBeforeMainModule: () => [
      require.resolve(
        path.resolve(__dirname, 'node_modules/react-native-macos/Libraries/Core/InitializeCore')),
      ]
  },
  transformer: {
    assetRegistryPath: path.resolve(__dirname, 'node_modules/react-native-macos/Libraries/Image/AssetRegistry'),
  },
```



For 0.62, so long as you are using @react-native-community/cli@4.9 or greater, then the cli should automatically set this up for you.
See https://github.com/react-native-community/cli/pull/1115 for details of the metro change that allowed this config to happen automatically.

**Comment by sidferreira** [comment]
@acoates-ms I just started a brand new project and this issue happens to me. I was using 0.63 first, but now I changed to 0.62.

**Comment by acoates-ms** [comment]
I was mistaken in my above comment about this happening automatically currently.  There was a change that I thought we'd made in react-native-macos that never happened.  If I work out what #587 needs, then it would work without a custom metro config.  Otherwise, you can run the snipped in my comment above to configure metro correctly for macos.

**Comment by sidferreira** [comment]
@acoates-ms Also, it seems that if you run `yarn ios` first and then `yarn macos` (or npx), it will fail cause `react-native start` from iOS won't include MacOS stuff. Now, if you run `macos` first, it will launch the modified Metro and work properly, including the `ios` build.

**Comment by acoates-ms** [comment]
Can you try this again with react-native-macos 0.62.11.

**Comment by Abhishek12345679** [comment]
It works but not with the command from the react-native-docs (npx react-native run-macos)

Instead it works when we start the metro server and app separately as -
1. yarn start:macos
2. npx react-native run-macos

**Comment by ejlayne** [comment]
@acoates-ms @HeyImChris FYI I ran into this again today, where I had run "yarn ios" from fluent-ui-react-native tester app, and then was trying to run "yarn macos" to have both testers up, but ran into this error. I don't know where documentation for react-native-macos is going, but if this issue can't be fixed, then we should mention it as a known issue in documentation (i.e. if you're running both ios and mac, make sure to run macos first). Let me know if you have any questions.

**Comment by KyCodes** [comment]
I ran into this again today, was able to get past the error with the code above,  but am now getting a "No bundle URL present" error. Any ideas?

**Comment by thertzelle** [comment]
I'm getting the same error from a clean install of everything:

```
error: Error: Unable to resolve module ./Libraries/Components/AccessibilityInfo/AccessibilityInfo from /Users/thertzelle/Local_Development/JQuakeMobile/node_modules/react-native/index.js: 

None of these files exist:
  * node_modules/react-native/Libraries/Components/AccessibilityInfo/AccessibilityInfo(.native|.macos.js|.native.js|.js|.macos.json|.native.json|.json|.macos.ts|.native.ts|.ts|.macos.tsx|.native.tsx|.tsx)
  * node_modules/react-native/Libraries/Components/AccessibilityInfo/AccessibilityInfo/index(.native|.macos.js|.native.js|.js|.macos.json|.native.json|.json|.macos.ts|.native.ts|.ts|.macos.tsx|.native.tsx|.tsx)
  ```

**Comment by thertzelle** [comment]
I re-installed my entire environment again, and followed setting up react native and then setting up react native macos, and this is the error I get.

**Comment by sungsong88** [comment]
Same here. I'm using an M1 MacBook air and same issue. I tried everything mentioned above.

**Comment by Bojianye** [comment]
Any update?
